server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -325,40 +334,59 @@ public static String executorName(Thread thread) {
         return executorName(thread.getName());
     }
 
-    public static ThreadFactory daemonThreadFactory(Settings settings, String namePrefix) {
-        return daemonThreadFactory(threadName(settings, namePrefix));
+    public static ThreadFactory daemonThreadFactory(Settings settings, String executorName) {
+        return createDaemonThreadFactory(threadName(settings, executorName), executorName);
     }
 
-    public static ThreadFactory daemonThreadFactory(String nodeName, String namePrefix) {
+    public static ThreadFactory daemonThreadFactory(String nodeName, String executorName) {
         assert nodeName != null && false == nodeName.isEmpty();
-        return daemonThreadFactory(threadName(nodeName, namePrefix));
+        return createDaemonThreadFactory(threadName(nodeName, executorName), executorName);
     }
 
-    public static ThreadFactory daemonThreadFactory(String namePrefix) {
-        return new EsThreadFactory(namePrefix);
+    public static ThreadFactory daemonThreadFactory(String name) {
+        assert name != null && name.isEmpty() == false;
+        return createDaemonThreadFactory(name, null);
+    }
+
+    private static ThreadFactory createDaemonThreadFactory(String namePrefix, String executorName) {
+        return new EsThreadFactory(namePrefix, executorName);
     }
 
     static class EsThreadFactory implements ThreadFactory {
 
         final ThreadGroup group;
         final AtomicInteger threadNumber = new AtomicInteger(1);
         final String namePrefix;
+        final boolean isSystem;
 
-        EsThreadFactory(String namePrefix) {
+        EsThreadFactory(String namePrefix, String executorName) {
             this.namePrefix = namePrefix;
             SecurityManager s = System.getSecurityManager();
             group = (s != null) ? s.getThreadGroup() : Thread.currentThread().getThreadGroup();
+            isSystem = executorName != null && SYSTEM_THREAD_PREFIXES.contains(executorName);
```

Comment:
I know I did this too in my prototype, but you have to admit it's a little lame? Think we could maybe pass the `isSystem` property through the steps of building the executor via a boolean flag?

Reply:
Makes sense - removed dependency on naming in 71a0d285f82aaa40698197df3fd90eaafce49480